assets/img/middleware-1.svg

```diff
@@ -0,0 +1 @@
9.90063376390456,0)" xlink:href="#L"/><use transform="matrix(0.05947632153027956,0,0,0.05947632153027956,61.79589806996048,0)" xlink:href="#J"/><use transform="matrix(0.05947632153027956,0,0,0.05947632153027956,73.69116237601638,0)" xlink:href="#ak"/><use transform="matrix(0.05947632153027956,0,0,0.05947632153027956,84.3969002514667,0)" xlink:href="#N"/></g><path fill="#333" d="M127-220V0H93v-220H8v-28h204v28h-85" id="aw"/><path fill="#333" d="M30-248c87 1 191-15 191 75 0 78-77 80-158 76V0H30v-248zm33 125c57 0 124 11 124-50 0-59-68-47-124-48v98" id="ax"/><g id="D"><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,0,0)" xlink:href="#ad"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,13.692770912304363,0)" xlink:href="#aw"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,25.270828170198783,0)" xlink:href="#aw"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,36.848885428093205,0)" xlink:href="#ax"/></g><g id="E"><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,0,0)" xlink:href="#as"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,10.5735682720497,0)" xlink:href="#at"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,21.1471365440994,0)" xlink:href="#at"/></g><path fill="#333" d="M140-251c81 0 123 46 123 126C263-46 219 4 140 4 59 4 17-45 17-125s42-126 123-126zm0 227c63 0 89-41 89-101s-29-99-89-99c-61 0-89 39-89 99S79-25 140-24" id="ay"/><path fill="#333" d="M194 0L95-120 63-95V0H30v-248h33v124l119-124h40L117-140 236 0h-42" id="az"/><g id="F"><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,0,0)" xlink:href="#ay"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,14.802995580869581,0)" xlink:href="#az"/></g><g id="G"><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,0,0)" xlink:href="#ax"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,12.688281926459641,0)" xlink:href="#ay"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,27.491277507329222,0)" xlink:href="#Z"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,40.17955943378887,0)" xlink:href="#aw"/></g><path fill="#333" d="M0 4l72-265h28L28 4H0" id="aA"/><g id="H"><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,0,0)" xlink:href="#aA"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,5.28678413602485,0)" xlink:href="#L"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,15.86035240807455,0)" xlink:href="#V"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,22.151625529944123,0)" xlink:href="#ac"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,32.725193801993825,0)" xlink:href="#N"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,43.29876207404352,0)" xlink:href="#V"/><use transform="matrix(0.0528678413602485,0,0,0.0528678413602485,49.590035195913096,0)" xlink:href="#aA"/></g></defs></g></svg>
```

Comment:
I feel like the middleware chain is a bit linear for a sequence diagram. If the process contained more asynchronous and parallel invocations it would work a bit better, but I think something simpler like a dataflow diagram could work better.

Reply:
I think that's fair enough, might be good to add a little bit of extra detail that can delineate the process a bit (currently every middleware either has a pre/post process related to the next middleware in the chain. 
![Capture](https://user-images.githubusercontent.com/13415171/137234819-6a540e3c-52a2-41d0-b714-390004e5e1bf.PNG)

The logging middleware example you have would be great illustration by adding a pre-next process as well as post-next process, e.g. Request & Response logging would be a pretty common example.

Adding a third-party auth provider might be good as well, but potentially borders on too much detail.